Sources/Java2SwiftLib/JavaTranslator.swift

```diff
@@ -458,6 +471,51 @@ extension JavaTranslator {
       """
   }
 
+  package func translateToEnumValue(name: String, enumFields: [Field]) -> [DeclSyntax] {
+    let extensionSyntax: DeclSyntax = """
+      public enum \(raw: name): Equatable {
+        \(raw: enumFields.map { "case \($0.getName())" }.joined(separator: "\n"))
+      }
+    """
+
+    let mappingSyntax: DeclSyntax = """
+      public var enumValue: \(raw: name)? {
+        let classObj = self.javaClass
+        \(raw: enumFields.map {
+          // The equals method takes a java object, so we need to cast it here
+          """
+          if self.equals(classObj.\($0.getName())?.as(JavaObject.self)) {
+                return \(name).\($0.getName())
+          }
+          """
+        }.joined(separator: " else ")) else {
+          return nil
+        }
+      }
+    """
+
+    let initSyntax: DeclSyntax = """
+    public init?(_ enumValue: \(raw: name), environment: JNIEnvironment) throws {
+      let classObj = try JavaClass<Self>(in: environment)
+      switch enumValue {
+    \(raw: enumFields.map {
+      let caseName = $0.getName()
+      return """
+          case .\(caseName):
```

Comment:
FYI this use of nested string interpolations triggered a Swift compiler crash in newly-introduced code in the compiler, which I've reduced and filed as https://github.com/swiftlang/swift/issues/77050. To avoid it (rather than waiting on a compiler fix), I think we can remove `caseName` and use `$0.getName()` everywhere.